plugins/foreman_ansible/1.x/index.md

```diff
@@ -87,8 +89,22 @@ The project is not packaged, but you can find the sources here at [theforeman/fo
 
 In Foreman, you should add whatever Ansible hosts you want to submit facts from to the setting `trusted_puppetmaster_hosts`. Change it at Administer > Settings, Auth tab.
 
+Ansible will run using the SSH key of the smart-proxy or Foreman. These keys are automatically set by Foreman Remote Execution, so you may see them at /usr/share/foreman-proxy/.ssh . The foreman-proxy user needs write access to '/usr/share/foreman-proxy/.ansible' at least to create the '.ansible' local directory. We are working on making the instller automatically do this, but if it didn't do it.
+
+```
+sudo mkdir /usr/share/foreman-proxy/.ansible
+sudo chown foreman-proxy.foreman-proxy /usr/share/foreman-proxy/.ansible
+sudo chown foreman-proxy.foreman-proxy /usr/share/foreman-proxy
+```
+
+You may need do the same for the `foreman` user, so that it can run Ansible directly from the Foreman host without a Foreman Proxy.
+
+If you have Foreman Remote Execution already, it is likely you have a ssh key in `/usr/share/foreman-proxy/.ssh/id_rsa_foreman_proxy`. You may copy it to `/usr/share/foreman-proxy/.ssh/id_rsa` so that Ansible is able to use it (ensure that the file is readable by the `foreman-proxy` user)
```

Comment:
does ansible require it to be named `id_rsa`? can't it also use `id_rsa_foreman_proxy`? and it might be useful to mentione that for ansible, the key can't be protected with passphrase.

Reply:
I tried using `ansible_private_key_file=/usr/share/foreman-proxy/.ssh/id_rsa_foreman_proxy` on `/usr/share/foreman-proxy/.ansible.cfg`, to no avail, if you know how to make that work that'd be a good thing to put in https://github.com/theforeman/puppet-foreman_proxy/blob/master/templates/plugin/ansible.cfg.erb

Reply:
I thought ansible would use any ssh key available to the user, like ssh does but I could be wrong, the option accroding to man should be `key_file` or private_key` so I'd try `ansible_private_key=...`

Reply:
and looking at forklift's ansible.cfg, I see `private_key_file = $HOME/.vagrant.d/insecure_private_key` so I guess it should be possible

Reply:
:man_facepalming: I just saw we're sending a `ansible_private_key_file = nil` on the inventory when the setting is empty. Therefore it just points to `~.ssh/id_rsa` by default. When I change `ansible_private_key_file` at the setting or at the host level, it's fine.

I should be documenting this here - for some reason the setting does not show up in Settings > Ansible, local blip or are you unable to see the setting too @ares ?

Reply:
I don't see it either even though I see the definition in ./app/models/setting/ansible.rb

Reply:
https://github.com/theforeman/foreman_ansible/pull/142